core/src/main/java/org/apache/accumulo/fate/util/Retry.java

```diff
@@ -67,6 +71,25 @@ private Retry(long maxRetries, long startWait, long waitIncrement, long maxWait,
     this.logIntervalNanoSec = MILLISECONDS.toNanos(logInterval);
     this.hasNeverLogged = true;
     this.lastRetryLog = -1;
+    this.waitFactor = waitFactor;
+    this.currentJitter = jitter;
+    /*
+     * The call to Math.random will give us a value from 1 to 50. This when converted to
+     * milliseconds in the sleep call should add some significant jitter to the start time of the
+     * retries.
+     */
+    this.currentJitter = jitter;
```

Comment:
If the wait interval was already small, this jitter could dominate. That's not what we want. It'd be better if it was proportional. The way to do this would be to include it in the waitFactor, as some decimal portion.

For example, if the exponential backoff wait factor was 1.5, then this jitter could add some randomness from 1.4 to 1.5.

In general, I think you can get plus or minus 5% of the waitFactor by doing something like:

```java
    currentWait = (1 + (rand.nextDouble() - 0.5)/10) * waitFactor;
```

The `rand.nextDouble() - 0.5` will give a random value in the interval `[-0.5, 0.5)`. Dividing by ten and adding 1, gives a random value in the interval `[0.95, 1.05)`, so each increment is between `95%` and `105%` of the waitFactor.